packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -608,44 +630,38 @@ contract Bridge is EssentialContract, IBridge {
         }
     }
 
-    function _invocationGasLimit(
-        Message calldata _message,
-        bool _checkThe63Over64Rule
+    ///@dev This implementation was suggested by OpenZeppelin in an audit to replace the previous.
+    function _hasInsufficientGas(
+        uint256 _minGas,
+        uint256 _dataLength
     )
         private
         view
-        returns (uint256 gasLimit_)
+        returns (bool result_)
     {
         unchecked {
-            uint256 minGasRequired = getMessageMinGasLimit(_message.data.length);
-            gasLimit_ = minGasRequired.max(_message.gasLimit) - minGasRequired;
-        }
+            // https://github.com/ethereum/execution-specs/blob/master/src/ethereum/cancun/vm/gas.py#L128
+            uint256 words = _dataLength / 32 + 1;
+            uint256 memoryGasCost = words * 3 + words * words / 512;
```

Comment:
I don't think we need to add this check here because we don't know if the memory expansion is required for this data inside the invocation call, and it's fully controlled by the user so nothing can be manipulated.

Also the `words * words / 512` part at least should be removed because that part should be done on the total memory size when it's done which is unknown here, so can't calculate it at this point.

Reply:
I misunderstood the problem and though this was about something else, if this is the initial memory expansion this is the correct formula.